handler/src/main/java/io/netty/handler/timeout/WriteTimeoutHandler.java

```diff
@@ -70,6 +70,8 @@
 
     private final long timeoutNanos;
 
+    private Set<ScheduledFuture<?>> timeouts;
```

Comment:
So it's guaranteed that ChannelPromise's listener will be called before the next `write` event?